### PR TITLE
Setting up CMake

### PR DESCRIPTION
**Pull request by EmperorYP7**
This is a very infant stage of the project. Docs are up and CMake is building the project successfully. Will be configuring compiler and linker settings as we make the app. We'll be using Poco library for handling HTTP requests and Google Test for testing. Both of them are added as a submodule in the project.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!-- Do NOT delete this template. Its great that you decided to contribute 
+to our project. :) Kindly fill the template accordingly! -->
+### Solves Issue # <number>
```
Comment (anktrj) [inline .github/PULL_REQUEST_TEMPLATE.md]: change this to `fixes #<num>` or `closes #<num>`

**Review by anktrj — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (EmperorYP7) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<!-- Do NOT delete this template. Its great that you decided to contribute 
+to our project. :) Kindly fill the template accordingly! -->
+### Fixes issue # <number>
```
Comment (anktrj) [inline .github/PULL_REQUEST_TEMPLATE.md]: again if you write `fixes #num` github automatically closes the issue and this doesn't happen with `Fixes issue #num`

**Review by anktrj — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (EmperorYP7) [reply .github/PULL_REQUEST_TEMPLATE.md]: Alright. Fixing it.

**Comment by EmperorYP7** [comment]
should we merge this now?

**Comment by EmperorYP7** [comment]
Merging this as per non-public clearance from the maintainers.

**Pull request merged**